src/Analyzers/Core/Analyzers/SimplifyInterpolation/AbstractHelpers.cs

```diff
@@ -160,17 +145,18 @@ private static TextSpan GetSpanWithinLiteralQuotes(IVirtualCharService virtualCh
                             IsSpaceChar(invocation.Arguments[1]))
                         {
                             var alignmentOp = invocation.Arguments[0].Value;
-                            if (alignmentOp != null && alignmentOp.ConstantValue.HasValue)
+                            if (alignmentOp != null && (PermitNonLiteralAlignmentComponents
+                                ? alignmentOp.ConstantValue.HasValue
+                                : alignmentOp.Kind == OperationKind.Literal))
```

Comment:
please break this up.  this is too complex for an if-conditional.  thx.

Reply:
Thanks. Can you give your opinion on the form I replaced it with?